server/zoom/oauth.go

```diff
@@ -151,25 +151,25 @@ func (c *OAuthClient) getUserViaOAuth(user *model.User) (*User, error) {
 
 		c.token = updatedToken
 	} else {
+		// This block is only for refreshing the user's OAuth token. If we failed to fetch the current token,
+		// it is likely that this is the first the time the user has connected, so we just avoid refreshing if there's an error.
```

Comment:
@mickmister - is there a way to detect this particular condition "it is likely that this is the **first the time** the user has connected" ? - like if that condition does not apply and there is an error, what can happen? Maybe open another ticket for later to follow-through on this and further refine the handling for this exact condition only?

Reply:
@catalintomai Thanks for pointing this out. Initially, I had modified the method `fetchOAuthUserInfo`, which is a more low-level method that fetches the token from the kv store, to signal to its caller when the token doesn't exist. At the moment, it returns an error both when an actual error occurs, and when the token just doesn't exist. I modified this method to return `nil, nil` if the token doesn't exist, then it is up to the caller to differentiate between the two situations (is `err` nil? if not is `token` nil?)

This logic would need to be applied to 5 places, all doing duplicate checking, and potentially causing a nil pointer panic in the future in a new spot that calls this method. Not to mention none of these particular callers need to care if it was an actual error, or the token is simply notexistent, besides this one case where we need to know that.

I was thinking explicitly passing in a boolean to signal "this is a first connect operation" would be cumbersome, or overly explicit. After reading your comment and thinking about it a bit more, I tried it out and it makes the code very readable and intentional. Not much code needs to change either.

I've updated the PR with this strategy, and am satisfied with the solution. Please take another look when you have the chance.

Reply:
@mickmister , thanks for following through, liking this much better! Prob. can update the comment as well.